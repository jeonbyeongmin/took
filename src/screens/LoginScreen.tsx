import {
  Box,
  Button,
  HStack,
  Icon,
  Input,
  NativeBaseProvider,
  Text,
  VStack,
} from 'native-base';
import React, {useState} from 'react';
import Ionicons from 'react-native-vector-icons/Ionicons';

function LoginScreen({navigation}: any) {
  const [show, setShow] = useState(false);
  return (
    <NativeBaseProvider>
      <Box h="100%" w="100%" paddingTop={20} backgroundColor="white">
        <VStack alignItems="center" justifyContent="center">
          <VStack space={6}>
            <Input
              w={96}
              h={12}
              // InputLeftElement={<Icon size={5} ml="2" color="muted.400" />}
              placeholder="이메일을 입력해주세요."
              fontSize="16px"
              variant="underlined"
            />
            <Input
              w={96}
              h={12}
              type={show ? 'text' : 'password'}
              // InputRightElement={}
              placeholder="비밀번호를 입력해주세요."
              fontSize="16px"
              variant="underlined"
            />
          </VStack>
          <VStack space={5} marginTop={16}>
            <Button
              colorScheme={'blue'}
              w={96}
              padding={'4'}
              onPress={() => console.log('hello world')}>
              <Text color={'white'} fontSize={'16px'}>
                로그인
              </Text>
            </Button>
            <Button
              variant="unstyled"
              w={'96'}
              padding={'4'}
              size="lg"
              onPress={() => navigation.navigate('Regist')}>
              회원가입
            </Button>
          </VStack>
          <VStack marginTop={16}>
            <Button
              backgroundColor={'light.100'}
              w={96}
              padding={4}
              onPress={() => console.log('google login')}>
              <HStack
                w={96}
                paddingX={4}
                alignItems={'center'}
                justifyContent="space-between">
                <Icon
                  as={Ionicons}
                  name="logo-google"
                  size="md"
                  color={'black'}
                />
                <Text fontSize={'16px'}>구글 아이디로 로그인</Text>
                <Icon
                  as={Ionicons}
                  name="chevron-forward-outline"
                  size={'md'}
                />
              </HStack>
            </Button>
          </VStack>
        </VStack>
      </Box>
    </NativeBaseProvider>
  );
}

export default LoginScreen;
