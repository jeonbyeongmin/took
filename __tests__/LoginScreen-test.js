import 'react-native';
import React from 'react';
import {fireEvent, render} from '@testing-library/react-native';
import LoginScreen from '@screens/Auth/LoginScreen';
import signInWithEmailAndPassword from '../__mocks__/@react-native-firebase/auth';
import {NativeBaseProvider} from 'native-base';

const inset = {
  frame: {x: 0, y: 0, width: 0, height: 0},
  insets: {top: 0, left: 0, right: 0, bottom: 0},
};

function getComponent(props) {
  return (
    <NativeBaseProvider initialWindowMetrics={inset}>
      <LoginScreen {...props} />
    </NativeBaseProvider>
  );
}

describe('App test...', () => {
  const props = {};
  const component = getComponent(props);
  test('test 2', () => {
    const rendering = render(component);
    expect(rendering).toMatchSnapshot();
    expect(rendering).toBeTruthy();
  });

  //   test('로그인에 성공하면 상태 코드 200을 반환받음', () => {
  //     const rendered = render(<LoginScreen {...props} />);

  //     fireEvent(rendered.getByText('로그인'), 'onPress');

  //     signInWithEmailAndPassword.mockImplementation(
  //       ({email, password}) => `${email}, ${password}`,
  //     );

  //     expect(signInWithEmailAndPassword).toBeCalledTimes(1);
  //     signInWithEmailAndPassword.mockReturnValue(
  //       Promise.resolve({
  //         statusCode: 200,
  //       }),
  //     );
  //   });
});
