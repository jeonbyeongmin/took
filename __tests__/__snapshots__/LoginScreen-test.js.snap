// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App test... test 2 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <View
    dataSet={Object {}}
    style={
      Object {
        "backgroundColor": "#FFFFFF",
        "height": "100%",
        "paddingTop": 80,
        "width": "100%",
      }
    }
  >
    <View
      dataSet={Object {}}
      style={
        Object {
          "alignItems": "center",
          "justifyContent": "center",
        }
      }
    >
      <View
        dataSet={Object {}}
        style={Object {}}
      >
        <View
          dataSet={Object {}}
          style={Object {}}
        >
          <View
            dataSet={Object {}}
            isFocused={false}
            style={
              Object {
                "alignItems": "center",
                "borderBottomWidth": 1,
                "borderColor": "#d4d4d4",
                "borderRadius": 0,
                "borderWidth": 0,
                "flexDirection": "row",
                "height": 48,
                "overflow": "hidden",
                "width": 384,
              }
            }
          >
            <TextInput
              accessible={true}
              dataSet={Object {}}
              editable={true}
              onBlur={[Function]}
              onChangeText={[Function]}
              onFocus={[Function]}
              onKeyPress={[Function]}
              placeholder="이메일을 입력해주세요"
              placeholderTextColor="#a3a3a3"
              secureTextEntry={false}
              selectionColor="#1f2937"
              style={
                Object {
                  "backgroundColor": "transparent",
                  "color": "#171717",
                  "flex": 1,
                  "fontFamily": undefined,
                  "fontSize": 16,
                  "fontStyle": "normal",
                  "fontWeight": "400",
                  "paddingBottom": 8,
                  "paddingLeft": 0,
                  "paddingRight": 12,
                  "paddingTop": 8,
                  "width": "100%",
                }
              }
            />
          </View>
        </View>
        <View
          dataSet={Object {}}
          style={
            Object {
              "height": 24,
            }
          }
        />
        <View
          dataSet={Object {}}
          style={Object {}}
        >
          <View
            dataSet={Object {}}
            isFocused={false}
            style={
              Object {
                "alignItems": "center",
                "borderBottomWidth": 1,
                "borderColor": "#d4d4d4",
                "borderRadius": 0,
                "borderWidth": 0,
                "flexDirection": "row",
                "height": 48,
                "overflow": "hidden",
                "width": 384,
              }
            }
          >
            <TextInput
              accessible={true}
              dataSet={Object {}}
              editable={true}
              onBlur={[Function]}
              onChangeText={[Function]}
              onFocus={[Function]}
              onKeyPress={[Function]}
              placeholder="비밀번호를 입력해주세요"
              placeholderTextColor="#a3a3a3"
              secureTextEntry={true}
              selectionColor="#1f2937"
              style={
                Object {
                  "backgroundColor": "transparent",
                  "color": "#171717",
                  "flex": 1,
                  "fontFamily": undefined,
                  "fontSize": 16,
                  "fontStyle": "normal",
                  "fontWeight": "400",
                  "paddingBottom": 8,
                  "paddingLeft": 0,
                  "paddingRight": 12,
                  "paddingTop": 8,
                  "width": "100%",
                }
              }
            />
          </View>
        </View>
      </View>
      <View
        dataSet={Object {}}
        style={Object {}}
      />
      <View
        dataSet={Object {}}
        style={
          Object {
            "marginTop": 64,
          }
        }
      >
        <View
          accessibilityRole="button"
          accessible={true}
          collapsable={false}
          colorScheme="blue"
          dataSet={Object {}}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#2563eb",
              "borderRadius": 4,
              "flexDirection": "row",
              "justifyContent": "center",
              "paddingBottom": 16,
              "paddingLeft": 16,
              "paddingRight": 16,
              "paddingTop": 16,
              "width": 384,
            }
          }
        >
          <View
            dataSet={Object {}}
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
              }
            }
            test={true}
          >
            <View
              dataSet={Object {}}
              style={Object {}}
            >
              <Text
                dataSet={Object {}}
                style={
                  Object {
                    "backgroundColor": undefined,
                    "color": "#FFFFFF",
                    "fontFamily": undefined,
                    "fontSize": 16,
                    "fontStyle": "normal",
                    "fontWeight": "400",
                    "letterSpacing": 0,
                    "lineHeight": 24,
                    "textDecorationLine": undefined,
                  }
                }
              >
                로그인
              </Text>
            </View>
          </View>
        </View>
        <View
          dataSet={Object {}}
          style={
            Object {
              "height": 20,
            }
          }
        />
        <View
          accessibilityRole="button"
          accessible={true}
          collapsable={false}
          colorScheme="primary"
          dataSet={Object {}}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "borderRadius": 4,
              "flexDirection": "row",
              "justifyContent": "center",
              "paddingBottom": 16,
              "paddingLeft": 16,
              "paddingRight": 16,
              "paddingTop": 16,
              "width": 384,
            }
          }
        >
          <View
            dataSet={Object {}}
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
              }
            }
            test={true}
          >
            <View
              dataSet={Object {}}
              style={Object {}}
            >
              <Text
                dataSet={Object {}}
                style={
                  Object {
                    "backgroundColor": undefined,
                    "color": "#000000",
                    "fontFamily": undefined,
                    "fontSize": 16,
                    "fontStyle": "normal",
                    "fontWeight": "400",
                    "letterSpacing": 0,
                    "lineHeight": 24,
                    "textDecorationLine": undefined,
                  }
                }
              >
                회원가입
              </Text>
            </View>
          </View>
        </View>
      </View>
      <View
        dataSet={Object {}}
        style={Object {}}
      />
      <View
        dataSet={Object {}}
        style={
          Object {
            "marginTop": 64,
          }
        }
      >
        <View
          accessibilityRole="button"
          accessible={true}
          collapsable={false}
          colorScheme="primary"
          dataSet={Object {}}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#f5f5f4",
              "borderRadius": 4,
              "flexDirection": "row",
              "justifyContent": "center",
              "paddingBottom": 16,
              "paddingLeft": 16,
              "paddingRight": 16,
              "paddingTop": 16,
              "width": 384,
            }
          }
        >
          <View
            dataSet={Object {}}
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
              }
            }
            test={true}
          >
            <View
              dataSet={Object {}}
              style={Object {}}
            >
              <View
                dataSet={Object {}}
                style={
                  Object {
                    "alignItems": "center",
                    "flexDirection": "row",
                    "justifyContent": "space-between",
                    "paddingLeft": 16,
                    "paddingRight": 16,
                    "width": 384,
                  }
                }
              >
                <Text
                  allowFontScaling={false}
                  dataSet={Object {}}
                  selectable={false}
                  style={
                    Array [
                      Object {
                        "color": undefined,
                        "fontSize": 12,
                      },
                      Object {
                        "color": "#000000",
                        "fontSize": 20,
                        "height": 20,
                        "lineHeight": 20,
                        "width": 20,
                      },
                      Object {
                        "fontFamily": "Ionicons",
                        "fontStyle": "normal",
                        "fontWeight": "normal",
                      },
                      Object {},
                    ]
                  }
                >
                  
                </Text>
                <View
                  dataSet={Object {}}
                  style={Object {}}
                />
                <Text
                  dataSet={Object {}}
                  style={
                    Object {
                      "backgroundColor": undefined,
                      "color": "#171717",
                      "fontFamily": undefined,
                      "fontSize": 16,
                      "fontStyle": "normal",
                      "fontWeight": "400",
                      "letterSpacing": 0,
                      "lineHeight": 24,
                      "textDecorationLine": undefined,
                    }
                  }
                >
                  구글 아이디로 로그인
                </Text>
                <View
                  dataSet={Object {}}
                  style={Object {}}
                />
                <Text
                  allowFontScaling={false}
                  dataSet={Object {}}
                  selectable={false}
                  style={
                    Array [
                      Object {
                        "color": undefined,
                        "fontSize": 12,
                      },
                      Object {
                        "color": "#737373",
                        "fontSize": 20,
                        "height": 20,
                        "lineHeight": 20,
                        "width": 20,
                      },
                      Object {
                        "fontFamily": "Ionicons",
                        "fontStyle": "normal",
                        "fontWeight": "normal",
                      },
                      Object {},
                    ]
                  }
                >
                  
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</RNCSafeAreaProvider>
`;
